### PR TITLE
HackStudio:  Add language detection from file basename

### PR DESCRIPTION
**Pull request by adamdb5**
Allows HackStudio to determine languages from filenames such as 'Makefile' and 'CMakeLists.txt'.

### CODE REVIEW

**Comment by BertalanD** [comment]
Please don't introduce merge commits.

If you want to update your local development branch to get the latest changes from master, run `git pull --rebase`.
If you want to change commits you've already pushed, use `git rebase -i` and the push with `git push --force`.

**Comment by adamdb5** [comment]
Sorry about the merge request, I was unsure how to remove it. Thanks for the help!

File: Userland/DevTools/HackStudio/Language.cpp [hunk 1]
```diff
@@ -30,6 +30,16 @@ Language language_from_file_extension(const String& extension)
     return Language::Unknown;
 }
 
+Language language_from_file_name(const String& name)
+{
+    if (name == "Makefile")
```
Comment (krkk) [inline Userland/DevTools/HackStudio/Language.cpp:35]: Maybe we also should add support for the *makefile*s (starting with a lowercase m) and *GNUmakefile*s, as these are also used by gnu make, and bsd make (but without the latter):

> If no -f option is present, make will look for the makefiles GNUmakefile, makefile, and Makefile, in that order.
https://manpages.debian.org/testing/make/make.1.en.html

> By default, the file makefile is used; if no such file is found, it tries Makefile.
https://man.openbsd.org/make

Also, an `.mk` extension can also be a makefile and it's used quite often: https://github.com/search?q=extension%3Amk (12M results)

**Comment by adamdb5** [comment]
^ Added support for "makefile", "Makefile", "GNUmakefile" and `.mk` extensions

**Pull request closed without merging**